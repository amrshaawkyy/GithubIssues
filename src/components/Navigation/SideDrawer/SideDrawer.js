import React from 'react';
import Logo from "../../Logo/Logo";
import NavigationItems from '../NavigationItems/NavigationItems';
import './SideDrawer.css';
import Backdrop from '../../UI/Backdrop/Backdrop';
import Aux from '../../../hoc/AUX1/Aux1';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
const sideDrawer = (props) => {
    let attachedClasses = ["SideDrawer", "Close"];
    if (props.open) {
        attachedClasses = ["SideDrawer", "Open"]
    }
    return (
        <Aux>
            <Backdrop show={props.open} clicked={props.closed} />
            <div className={attachedClasses.join(' ')}>
                <div className="LogoSide">
                    <Link to='/'>
                        <Logo /*height="80%"*/ />
                    </Link>
                </div>
                <nav>
                    <NavigationItems />
                </nav>
            </div>
        </Aux>
    );
};
sideDrawer.propTypes = {
    open: PropTypes.bool,
    closed: PropTypes.func
};
export default sideDrawer;