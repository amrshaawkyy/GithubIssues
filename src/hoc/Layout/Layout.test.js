import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import  Layout  from './Layout';
import SideDrawer from '../../components/Navigation/SideDrawer/SideDrawer';

configure({adapter: new Adapter()});

describe('<Layout />', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Layout />);
    });
    afterEach(() => {
        expect(wrapper).toMatchSnapshot();
    });
    it('should render <SideDrawer /> when showSideDrawer is true', () => {
        wrapper.setState({showSideDrawer: true});
        expect(wrapper.find(SideDrawer).length).toBe(1);
    });
});


