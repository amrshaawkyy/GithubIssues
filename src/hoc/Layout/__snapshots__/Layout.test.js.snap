// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render <SideDrawer /> when showSideDrawer is true 1`] = `ShallowWrapper {}`;
